Viewer improvements

davidrwilliamson opened the pull request:
Aimed at closing issue #32

tankefugl commented on pysilcam/silcamgui/guicalcs.py:
```diff
@@ -232,6 +232,11 @@ def liveview(datadir, config_file):
     import pysilcam.silcamgui.liveviewer as lv
     lv.liveview(datadir, config_file)
 
+def annotate(datadir, filename):
```
E302: expected 2 blank lines, found 1  <!-- Automatically posted by Lintly -->

tankefugl commented on pysilcam/silcamgui/guicalcs.py:
```diff
@@ -263,6 +269,8 @@ def silcview(datadir):
                 if event.key == pygame.K_RIGHT:
                     direction = 1
+                if event.key == pygame.K_a:
+                    annotate(datadir, f)
```
F821: undefined name 'f'  <!-- Automatically posted by Lintly -->

tankefugl replied:
F821: undefined name 'f'  <!-- Automatically posted by Lintly -->

tankefugl commented on pysilcam/silcamgui/guicalcs.py:
```diff
@@ -280,13 +288,13 @@ def silcview(datadir):
            im = pygame.transform.scale(im, size)
            screen.blit(im,(0,0))
-           label = font.render('ZOOM [F]: OFF', 1, (255, 255, 0))
+           label = font.render('ZOOM [F]: OFF', 1, font_colour)
```
E111: indentation is not a multiple of four  <!-- Automatically posted by Lintly -->

tankefugl commented on pysilcam/silcamgui/guicalcs.py:
```diff
@@ -295,25 +303,25 @@ def silcview(datadir):
         screen.blit(label,(0,0))
         label = font.render('Display FPS:' + str(c.get_fps()),
-                1, (255, 255, 0))
+                1, font_colour)
```
E128: continuation line under-indented for visual indent  <!-- Automatically posted by Lintly -->

tankefugl commented on pysilcam/silcamgui/liveviewer.py:
```diff
@@ -158,20 +161,20 @@ def liveview(datapath = '/mnt/DATA/emlynd/DATA/', config_filename = 'config_hard
         pygame.display.set_caption('Image display')
         label = font.render(str(timestamp) + '    Disp. FPS: ' +
-                str(c.get_fps()), 20, (255, 255, 0))
+                str(c.get_fps()), 20, font_colour)
```
E128: continuation line under-indented for visual indent  <!-- Automatically posted by Lintly -->

tankefugl replied:
E128: continuation line under-indented for visual indent  <!-- Automatically posted by Lintly -->

tankefugl commented on pysilcam/silcamgui/liveviewer.py:
```diff
@@ -158,20 +161,20 @@ def liveview(datapath = '/mnt/DATA/emlynd/DATA/', config_filename = 'config_hard
         screen.blit(label,(0,0))
         label = font.render('Esc to exit',
-                1, (255, 255, 0))
+                1, font_colour)
```
E128: continuation line under-indented for visual indent  <!-- Automatically posted by Lintly -->

tankefugl replied:
E128: continuation line under-indented for visual indent  <!-- Automatically posted by Lintly -->

tankefugl commented on pysilcam/silcamgui/guicalcs.py:
```diff
@@ -232,18 +232,24 @@ def liveview(datadir, config_file):
     import pysilcam.silcamgui.liveviewer as lv
     lv.liveview(datadir, config_file)
 
+def annotate(datadir, filename):
```
E302: expected 2 blank lines, found 1  <!-- Automatically posted by Lintly -->

davidrwilliamson commented:
What this PR now addresses:
* Fixed the inconsistent pause behaviour in the guicalcs file playback viewer
* Added barebones but functional annotation: press 'a' while paused and the currently shown filename will be written to annotations.txt in the same folder as the data being viewed.
* Whitespace issues that were upsetting the linter
* Text colour is now set in one place in both guicalcs and liveviewer, and I've set it to a turquoise that is more readable on light images

What it doesn't address:
* Harmonising or merging the 3 or 4 different viewers that exist in the PySilCam code
* RGB8/Bayer8 issues
* Dodgy viewer window sizing

davidrwilliamson commented:
Is there some way to rerun the lint check? It seems like it failed due to an error with Github rather than a problem with the code, and didn't report any status as a result.

emlynjdavies commented:
If you have the actions tab you can rerun the test from there. 

Otherwise push a small change like a space or . For example
